packages/shared/src/contexts/BootProvider.tsx

```diff
@@ -131,15 +131,21 @@ export const BootDataProvider = ({
   };
 
   const [initialLoad, setInitialLoad] = useState<boolean>(null);
-  const [cachedBootData, setCachedBootData] = useState<Partial<Boot>>(() => {
+  const [cachedBootData, setCachedBootData] = useState<Partial<Boot>>();
+
+  useEffect(() => {
     if (localBootData) {
-      return localBootData;
+      setCachedBootData(localBootData);
+
+      return;
     }
```

Comment:
So yeah, this adds another re-render top level to boot but it allows us to avoid hydration issues due to user.

Let's look at the simple example in JSX:
```jsx
const Component = ({ user }) => {
  return user ? 'a' : 'b'
}
```

What happens now:
1. server side user is `undefined` so we render `b`
2. client side user gets loaded inside useMemo default value function from local storage
3. client side renders `a` because user is defined (from local storage)
4. hydration error because `b` does not match `a` 
5. react has to re-render everything client side (affects performance)

What happens with this change of adding a top level re-render
1. server side user is `undefined` so we render `b`
2. client side user is still `undefined`, `useEffect` run only after first client side render 
3. hydration passes `b` matches `b`
4. `useEffect` executes, user is no longer `undefined`
5. app re-renders
6. react only re-paints to DOM elements related to `user` change